sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -123,6 +126,24 @@ private[hive] object HiveShim {
     // for Serialization
     def this() = this(null)
 
+    override def hashCode(): Int = {
+        if (instance != null && instance.isInstanceOf[GenericUDFMacro]) {
```

Comment:
@yhuai Ah.. thanks a lot. I understand now that its not possible for users to extend the macro.I will make the change.